### ISSUE TITLE
DataStore not successfully syncing to cloud

### ISSUE THREAD

**Issue by jaimemarijke** [aws-amplify/amplify-js#11173]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React Native

### Amplify APIs

GraphQL API

### Amplify Categories

storage, api

### Environment information

<details>

```
# Put output below this line
System:
    OS: macOS 13.2.1
    CPU: (10) x64 Apple M2 Pro
    Memory: 310.54 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.8.0 - ~/.nvm/versions/node/v18.8.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v18.8.0/bin/yarn
    npm: 9.6.3 - ~/.nvm/versions/node/v18.8.0/bin/npm
    Watchman: 2023.03.13.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 111.0.5563.146
    Firefox: 66.0.2
    Safari: 16.3
  npmPackages:
    @babel/core: ^7.20.0 => 7.21.4
    @babel/eslint-parser: ^7.21.3 => 7.21.3
    @babel/runtime: ^7.21.0 => 7.21.0
    @expo/vector-icons: ^13.0.0 => 13.0.0
    @react-native-async-storage/async-storage: 1.17.11 => 1.17.11
    @react-native-community/datetimepicker: 6.7.3 => 6.7.3
    @react-native-community/masked-view: ^0.1.11 => 0.1.11
    @react-native-community/netinfo: 9.3.7 => 9.3.7
    @react-navigation/elements: ^1.3.17 => 1.3.17
    @react-navigation/native: ^6.1.6 => 6.1.6
    @react-navigation/native-stack: ^6.9.12 => 6.9.12
    @testing-library/jest-native: ^5.4.2 => 5.4.2
    @testing-library/react-native: ^12.0.1 => 12.0.1
    HelloWorld:  0.0.1
    ReactNativeRootToast:  0.0.1
    amazon-cognito-identity-js: ^6.2.0 => 6.2.0
    aws-amplify: ^5.0.23 => 5.0.23
    eslint: ^8.37.0 => 8.37.0
    eslint-config-airbnb: ^19.0.4 => 19.0.4
    eslint-config-prettier: ^8.8.0 => 8.8.0
    eslint-plugin-import: ^2.27.5 => 2.27.5
    eslint-plugin-jest: ^27.2.1 => 27.2.1
    eslint-plugin-jsx-a11y: ^6.7.1 => 6.7.1
    eslint-plugin-prettier: ^4.2.1 => 4.2.1
    eslint-plugin-react: ^7.32.2 => 7.32.2
    eslint-plugin-react-hooks: ^4.6.0 => 4.6.0
    expo: ~48.0.9 => 48.0.9
    expo-asset: ~8.9.1 => 8.9.1
    expo-camera: ~13.2.1 => 13.2.1
    expo-constants: ~14.2.1 => 14.2.1
    expo-dev-client: ~2.1.6 => 2.1.6
    expo-font: ~11.1.1 => 11.1.1
    expo-image-manipulator: ~11.1.1 => 11.1.1
    expo-location: ~15.1.1 => 15.1.1
    expo-splash-screen: ~0.18.1 => 0.18.1
    expo-status-bar: ~1.4.4 => 1.4.4
    expo-updates: ~0.16.3 => 0.16.3
    husky: ^8.0.3 => 8.0.3
    jest: ^29.5.0 => 29.5.0
    jest-expo: ^48.0.2 => 48.0.2
    lint-staged: ^13.2.0 => 13.2.0
    prettier: ^2.8.7 => 2.8.7
    prop-types: ^15.8.1 => 15.8.1
    react: 18.2.0 => 18.2.0 (16.14.0)
    react-native: 0.71.4 => 0.71.4
    react-native-gesture-handler: ~2.9.0 => 2.9.0
    react-native-logs: ^5.0.1 => 5.0.1
    react-native-maps: 1.3.2 => 1.3.2
    react-native-nfc-manager: ^3.14.2 => 3.14.2
    react-native-paper: 4.9.2 => 4.9.2
    react-native-reanimated: ~2.14.4 => 2.14.4
    react-native-root-siblings: ^4.1.1 => 4.1.1
    react-native-root-toast: ^3.4.1 => 3.4.1
    react-native-safe-area-context: 4.5.0 => 4.5.0
    react-native-screens: ~3.20.0 => 3.20.0
    react-native-uuid: ^2.0.1 => 2.0.1
    react-test-renderer: ^18.2.0 => 18.2.0
  npmGlobalPackages:
    @aws-amplify/cli: 10.6.1
    corepack: 0.13.0
    eas-cli: 3.8.1
    eslint-plugin-prettier: 4.2.1
    npm: 9.6.3
    yarn: 1.22.19

```

</details>

### Describe the bug

Possibly a dupe of https://github.com/aws-amplify/amplify-js/issues/8928

After upgrading to Expo SDK v48 (and associated upgrade to React Native 0.71.4), I started having issues where my data was not synced to my AWS DynamoDB. Though every once in a while it seems like one record gets synced.

Possibly related: when I try to view that same data in the Amplify Studio Data Manager, I don't see any records. Or every once in a while I see 1-3 records, but there are 40 or so records visible when I view the table directly in DynamoDB.

In the amplify debug logs, I see this error: `TypeError: null is not an object (evaluating '_c[opName].items')`
```
 LOG  [DEBUG] 22:37.173 Util -  attempt #11 with this vars: ["query operation($limit: Int, $nextToken: String, $lastSync: AWSTimestamp, $filter: ModelTagRecordFilterInput){\n\t\tsyncTagRecords(limit: $limit, nextToken: $nextToken, lastSync: $lastSync, filter: $filter){\n\t\t\titems {\n\t\t\t\t\t\t\tid\ntagId\ndateTime\nforkLength\nspecies\nlatitude\nlongitude\nphotoBase64\ncreatedAt\nupdatedAt\n_version\n_lastChangedAt\n_deleted\n\t\t\t\t\t\t}\n\t\t\t\t\t\tnextToken\n\t\t\t\t\t\tstartedAt\n\t\t}\n\t}",{"limit":1000,"nextToken":null,"lastSync":0,"filter":null}]
 LOG  [DEBUG] 22:37.178 RestClient - POST https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql
 LOG  [DEBUG] 22:37.881 Util - error on  [TypeError: null is not an object (evaluating '_c[opName].items')]
 LOG  [DEBUG] 22:37.884 Util -  retrying in 204887.13716594744 ms
```

### Expected behavior

Records sync from my app to the DynamoDB in the cloud

### Reproduction steps

Take an app that is successfully syncing data
Upgrade from Expo SDK 47->48, and React Native 0.70.5 -> 0.71.4
See that the app stops syncing data

### Code Snippet

The code has been unchanged - the only thing that changed was package versions

```javascript
// Put your code below this line.
const submitData = (tagRecordData) => {
  let errorMessage;
  try {
    LOG.debug('Attempting to create new TagRecord');
    LOG.debug({ ...tagRecordData, photoBase64: null });
    DataStore.save(new TagRecord({ ...tagRecordData, id: uuid.v4() }));
  } catch (error) {
    LOG.error(error);
    errorMessage = `Unable to save data, please try again. ${error.message}`;
  }
  return errorMessage;
};


```

### Log output

I set `Amplify.Logger.LOG_LEVEL = 'DEBUG';` in my app - here are the DEBUG logs

<details>

```
// Put your logs below this line
 LOG  4:19:08 PM | DEBUG :
{
  "id": null,
  "tagId": "HT-12345-Stub",
  "dateTime": "2023-03-31T23:18:57.029Z",
  "forkLength": 4.19,
  "species": "Alligator Gar",
  "latitude": 48.74,
  "longitude": -122.447,
  "photoBase64": null
}
 LOG  [DEBUG] 19:08.337 DataStore - Starting DataStore
 LOG  [DEBUG] 19:08.338 DataStore - Starting Storage
 LOG  [DEBUG] 19:08.376 DataStore - GraphQL endpoint available https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql
 LOG  [INFO] 19:08.380 DataStore - starting sync engine...
 LOG  [INFO] 19:08.401 Reachability - subscribing to reachability in React Native
 LOG  [INFO] 19:08.530 Reachability - Notifying reachability change true
 LOG  [DEBUG] 19:08.531 AuthClass - getting current credentials
 LOG  [DEBUG] 19:08.531 Credentials - getting credentials
 LOG  [DEBUG] 19:08.532 Credentials - picking up credentials
 LOG  [DEBUG] 19:08.532 Credentials - getting new cred promise
 LOG  [DEBUG] 19:08.532 Credentials - checking if credentials exists and not expired
 LOG  [DEBUG] 19:08.532 Credentials - need to get a new credential or refresh the existing one
 LOG  [DEBUG] 19:08.532 Credentials - no credentials for expiration check
 LOG  [DEBUG] 19:08.532 AuthClass - Getting current user credentials
 LOG  [DEBUG] 19:08.533 AuthClass - failed to get or parse item aws-amplify-federatedInfo [SyntaxError: JSON Parse error: Unexpected identifier "undefined"]
 LOG  [DEBUG] 19:08.533 AuthClass - Getting current session
 LOG  [DEBUG] 19:08.533 AuthClass - getting guest credentials
 LOG  [DEBUG] 19:08.534 Credentials - setting credentials for guest
 LOG  [DEBUG] 19:08.534 Parser - parse config [{"aws_appsync_apiKey": "da2-jo44siqhjjhjdohti2drqw2uvi", "aws_appsync_authenticationType": "API_KEY", "aws_appsync_graphqlEndpoint": "https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql", "aws_appsync_region": "us-west-2", "aws_project_region": "us-west-2", "identityPoolId": undefined, "identityPoolRegion": undefined, "mandatorySignIn": undefined, "refreshHandlers": undefined, "region": undefined, "storage": [Function MemoryStorage], "userPoolId": undefined}, "to amplifyconfig", {"Analytics": {}, "Auth": {}, "Logging": {}, "Storage": {"aws_appsync_apiKey": "da2-jo44siqhjjhjdohti2drqw2uvi", "aws_appsync_authenticationType": "API_KEY", "aws_appsync_graphqlEndpoint": "https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql", "aws_appsync_region": "us-west-2", "aws_project_region": "us-west-2", "identityPoolId": undefined, "identityPoolRegion": undefined, "mandatorySignIn": undefined, "refreshHandlers": undefined, "region": undefined, "storage": [Function MemoryStorage], "userPoolId": undefined}}]
 LOG  [DEBUG] 19:08.535 Credentials - No Cognito Identity pool provided for unauthenticated access
 LOG  [DEBUG] 19:08.536 AuthClass - Getting current session
 LOG  [DEBUG] 19:08.536 DataStore - error getting OIDC JWT Auth is not configured
 LOG  [INFO] 19:08.537 DataStore - Realtime ready
 LOG  [DEBUG] 19:08.538 DataStore - Attempting Create subscription with authMode: API_KEY
 LOG  [DEBUG] 19:08.539 PubSub - subscribe options {"additionalHeaders": {}, "apiKey": "da2-jo44siqhjjhjdohti2drqw2uvi", "appSyncGraphqlEndpoint": "https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql", "authToken": undefined, "authenticationType": "API_KEY", "graphql_headers": [Function anonymous], "provider": Symbol(INTERNAL_AWS_APPSYNC_REALTIME_PUBSUB_PROVIDER), "query": "subscription operation {
  onCreateTagRecord {
    id
    tagId
    dateTime
    forkLength
    species
    latitude
    longitude
    photoBase64
    createdAt
    updatedAt
    _version
    _lastChangedAt
    _deleted
  }
}
", "region": "us-west-2", "variables": {}}
 LOG  [INFO] 19:08.540 Reachability - subscribing to reachability in React Native
 LOG  [INFO] 19:08.540 Reachability - Notifying reachability change true
 LOG  [DEBUG] 19:08.541 AWSAppSyncRealTimeProvider - Authenticating with API_KEY
 LOG  [DEBUG] 19:08.542 DataStore - Attempting Update subscription with authMode: API_KEY
 LOG  [DEBUG] 19:08.542 PubSub - subscribe options {"additionalHeaders": {}, "apiKey": "da2-jo44siqhjjhjdohti2drqw2uvi", "appSyncGraphqlEndpoint": "https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql", "authToken": undefined, "authenticationType": "API_KEY", "graphql_headers": [Function anonymous], "provider": Symbol(INTERNAL_AWS_APPSYNC_REALTIME_PUBSUB_PROVIDER), "query": "subscription operation {
  onUpdateTagRecord {
    id
    tagId
    dateTime
    forkLength
    species
    latitude
    longitude
    photoBase64
    createdAt
    updatedAt
    _version
    _lastChangedAt
    _deleted
  }
}
", "region": "us-west-2", "variables": {}}
 LOG  [DEBUG] 19:08.542 AWSAppSyncRealTimeProvider - Authenticating with API_KEY
 LOG  [DEBUG] 19:08.542 DataStore - Attempting Delete subscription with authMode: API_KEY
 LOG  [DEBUG] 19:08.543 PubSub - subscribe options {"additionalHeaders": {}, "apiKey": "da2-jo44siqhjjhjdohti2drqw2uvi", "appSyncGraphqlEndpoint": "https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql", "authToken": undefined, "authenticationType": "API_KEY", "graphql_headers": [Function anonymous], "provider": Symbol(INTERNAL_AWS_APPSYNC_REALTIME_PUBSUB_PROVIDER), "query": "subscription operation {
  onDeleteTagRecord {
    id
    tagId
    dateTime
    forkLength
    species
    latitude
    longitude
    photoBase64
    createdAt
    updatedAt
    _version
    _lastChangedAt
    _deleted
  }
}
", "region": "us-west-2", "variables": {}}
 LOG  [DEBUG] 19:08.543 AWSAppSyncRealTimeProvider - Authenticating with API_KEY
 LOG  [DEBUG] 19:08.543 Hub - Dispatching to api with  {"data": {"connectionState": "Disconnected", "provider": {"_config": [Object], "connectionStateMonitor": [ConnectionStateMonitor], "connectionStateMonitorSubscription": [Subscription], "keepAliveTimeout": 300000, "promiseArray": [Array], "reconnectionMonitor": [ReconnectionMonitor], "socketStatus": 0, "subscriptionObserverMap": [Map]}}, "event": "ConnectionStateChange", "message": "Connection state is Disconnected"}
 LOG  [DEBUG] 19:08.544 Hub - Dispatching to api with  {"data": {"connectionState": "Disconnected", "provider": {"_config": [Object], "connectionStateMonitor": [ConnectionStateMonitor], "connectionStateMonitorSubscription": [Subscription], "keepAliveTimeout": 300000, "promiseArray": [Array], "reconnectionMonitor": [ReconnectionMonitor], "socketStatus": 0, "subscriptionObserverMap": [Map]}}, "event": "ConnectionStateChange", "message": "Connection state is Disconnected"}
 LOG  [DEBUG] 19:08.546 Hub - Dispatching to api with  {"data": {"connectionState": "Disconnected", "provider": {"_config": [Object], "connectionStateMonitor": [ConnectionStateMonitor], "connectionStateMonitorSubscription": [Subscription], "keepAliveTimeout": 300000, "promiseArray": [Array], "reconnectionMonitor": [ReconnectionMonitor], "socketStatus": 0, "subscriptionObserverMap": [Map]}}, "event": "ConnectionStateChange", "message": "Connection state is Disconnected"}
 LOG  [INFO] 19:08.547 Reachability - subscribing to reachability in React Native
 LOG  [INFO] 19:08.547 Reachability - Notifying reachability change true
 LOG  [DEBUG] 19:08.547 Hub - Dispatching to api with  {"data": {"connectionState": "Connecting", "provider": {"_config": [Object], "connectionState": "Disconnected", "connectionStateMonitor": [ConnectionStateMonitor], "connectionStateMonitorSubscription": [Subscription], "keepAliveTimeout": 300000, "promiseArray": [Array], "reconnectionMonitor": [ReconnectionMonitor], "socketStatus": 0, "subscriptionObserverMap": [Map]}}, "event": "ConnectionStateChange", "message": "Connection state is Connecting"}
 LOG  [DEBUG] 19:08.548 Hub - Dispatching to api with  {"data": {"connectionState": "Connecting", "provider": {"_config": [Object], "connectionState": "Disconnected", "connectionStateMonitor": [ConnectionStateMonitor], "connectionStateMonitorSubscription": [Subscription], "keepAliveTimeout": 300000, "promiseArray": [Array], "reconnectionMonitor": [ReconnectionMonitor], "socketStatus": 0, "subscriptionObserverMap": [Map]}}, "event": "ConnectionStateChange", "message": "Connection state is Connecting"}
 LOG  [DEBUG] 19:08.549 Hub - Dispatching to api with  {"data": {"connectionState": "Connecting", "provider": {"_config": [Object], "connectionState": "Disconnected", "connectionStateMonitor": [ConnectionStateMonitor], "connectionStateMonitorSubscription": [Subscription], "keepAliveTimeout": 300000, "promiseArray": [Array], "reconnectionMonitor": [ReconnectionMonitor], "socketStatus": 0, "subscriptionObserverMap": [Map]}}, "event": "ConnectionStateChange", "message": "Connection state is Connecting"}
 LOG  [DEBUG] 19:08.550 AWSAppSyncRealTimeProvider - Authenticating with API_KEY
 LOG  [DEBUG] 19:08.551 AWSAppSyncRealTimeProvider - Initializaling retryable Handshake
 LOG  [DEBUG] 19:08.551 Util - bound  attempt #1 with this vars: ["wss://vnyaearw6zcbllzmjv2ti2w75u.appsync-realtime-api.us-west-2.amazonaws.com/graphql?header=eyJob3N0Ijoidm55YWVhcnc2emNibGx6bWp2MnRpMnc3NXUuYXBwc3luYy1hcGkudXMtd2VzdC0yLmFtYXpvbmF3cy5jb20iLCJ4LWFtei1kYXRlIjoiMjAyMzAzMzFUMjMxOTA4WiIsIngtYXBpLWtleSI6ImRhMi1qbzQ0c2lxaGpqaGpkb2h0aTJkcnF3MnV2aSJ9&payload=e30="]
 LOG  [DEBUG] 19:08.551 AWSAppSyncRealTimeProvider - Initializing handshake wss://vnyaearw6zcbllzmjv2ti2w75u.appsync-realtime-api.us-west-2.amazonaws.com/graphql?header=eyJob3N0Ijoidm55YWVhcnc2emNibGx6bWp2MnRpMnc3NXUuYXBwc3luYy1hcGkudXMtd2VzdC0yLmFtYXpvbmF3cy5jb20iLCJ4LWFtei1kYXRlIjoiMjAyMzAzMzFUMjMxOTA4WiIsIngtYXBpLWtleSI6ImRhMi1qbzQ0c2lxaGpqaGpkb2h0aTJkcnF3MnV2aSJ9&payload=e30=
 LOG  [DEBUG] 19:08.553 DataStore - Attempting sync with authMode: API_KEY
 LOG  [DEBUG] 19:08.555 Util -  attempt #1 with this vars: ["query operation($limit: Int, $nextToken: String, $lastSync: AWSTimestamp, $filter: ModelTagRecordFilterInput){\n\t\tsyncTagRecords(limit: $limit, nextToken: $nextToken, lastSync: $lastSync, filter: $filter){\n\t\t\titems {\n\t\t\t\t\t\t\tid\ntagId\ndateTime\nforkLength\nspecies\nlatitude\nlongitude\nphotoBase64\ncreatedAt\nupdatedAt\n_version\n_lastChangedAt\n_deleted\n\t\t\t\t\t\t}\n\t\t\t\t\t\tnextToken\n\t\t\t\t\t\tstartedAt\n\t\t}\n\t}",{"limit":1000,"nextToken":null,"lastSync":0,"filter":null}]
 LOG  [DEBUG] 19:08.561 RestClient - POST https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql
 LOG  [DEBUG] 19:08.829 AWSAppSyncRealTimeProvider - subscription message from AWS AppSyncRealTime: {"type":"connection_ack","payload":{"connectionTimeoutMs":300000}}
 LOG  [DEBUG] 19:08.832 AWSAppSyncRealTimeProvider - Notifying connection successful
 LOG  [DEBUG] 19:08.833 AWSAppSyncRealTimeProvider - Notifying connection successful
 LOG  [DEBUG] 19:08.833 AWSAppSyncRealTimeProvider - Notifying connection successful
 LOG  [DEBUG] 19:08.837 AWSAppSyncRealTimeProvider - subscription message from AWS AppSync RealTime: {"type":"ka"}
 LOG  [DEBUG] 19:08.837 AWSAppSyncRealTimeProvider {"id": "", "observer": null, "query": "", "variables": {}}
 LOG  [DEBUG] 19:08.969 AWSAppSyncRealTimeProvider - subscription message from AWS AppSync RealTime: {"id":"61c6be6b-b4f6-4322-ac4c-71e4e44c8b1a","type":"start_ack"}
 LOG  [DEBUG] 19:08.970 AWSAppSyncRealTimeProvider {"id": "61c6be6b-b4f6-4322-ac4c-71e4e44c8b1a", "observer": {"_subscription": {"_cleanup": [Function anonymous], "_observer": [Object], "_queue": undefined, "_state": "ready"}}, "query": "subscription operation {
  onCreateTagRecord {
    id
    tagId
    dateTime
    forkLength
    species
    latitude
    longitude
    photoBase64
    createdAt
    updatedAt
    _version
    _lastChangedAt
    _deleted
  }
}
", "variables": {}}
 LOG  [DEBUG] 19:08.972 AWSAppSyncRealTimeProvider - subscription ready for {"query":"subscription operation {\n  onCreateTagRecord {\n    id\n    tagId\n    dateTime\n    forkLength\n    species\n    latitude\n    longitude\n    photoBase64\n    createdAt\n    updatedAt\n    _version\n    _lastChangedAt\n    _deleted\n  }\n}\n","variables":{}}
 LOG  [DEBUG] 19:08.973 Hub - Dispatching to api with  {"data": {"query": "subscription operation {
  onCreateTagRecord {
    id
    tagId
    dateTime
    forkLength
    species
    latitude
    longitude
    photoBase64
    createdAt
    updatedAt
    _version
    _lastChangedAt
    _deleted
  }
}
", "variables": {}}, "event": "Subscription ack", "message": "Connection established for subscription"}
 LOG  [DEBUG] 19:08.975 Hub - Dispatching to api with  {"data": {"query": "subscription operation {
  onCreateTagRecord {
    id
    tagId
    dateTime
    forkLength
    species
    latitude
    longitude
    photoBase64
    createdAt
    updatedAt
    _version
    _lastChangedAt
    _deleted
  }
}
", "variables": {}}, "event": "Subscription ack", "message": "Connection established for subscription"}
 LOG  [DEBUG] 19:08.976 Hub - Dispatching to api with  {"data": {"query": "subscription operation {
  onCreateTagRecord {
    id
    tagId
    dateTime
    forkLength
    species
    latitude
    longitude
    photoBase64
    createdAt
    updatedAt
    _version
    _lastChangedAt
    _deleted
  }
}
", "variables": {}}, "event": "Subscription ack", "message": "Connection established for subscription"}
 LOG  [DEBUG] 19:08.977 Hub - Dispatching to api with  {"data": {"connectionState": "Connected", "provider": {"_config": [Object], "awsRealTimeSocket": [WebSocket], "connectionState": "Connecting", "connectionStateMonitor": [ConnectionStateMonitor], "connectionStateMonitorSubscription": [Subscription], "keepAliveAlertTimeoutId": 738, "keepAliveTimeout": 300000, "keepAliveTimeoutId": 737, "promiseArray": [Array], "reconnectionMonitor": [ReconnectionMonitor], "socketStatus": 1, "subscriptionObserverMap": [Map]}}, "event": "ConnectionStateChange", "message": "Connection state is Connected"}
 LOG  [DEBUG] 19:08.984 Hub - Dispatching to api with  {"data": {"connectionState": "Connected", "provider": {"_config": [Object], "awsRealTimeSocket": [WebSocket], "connectionState": "Connecting", "connectionStateMonitor": [ConnectionStateMonitor], "connectionStateMonitorSubscription": [Subscription], "keepAliveAlertTimeoutId": 738, "keepAliveTimeout": 300000, "keepAliveTimeoutId": 737, "promiseArray": [Array], "reconnectionMonitor": [ReconnectionMonitor], "socketStatus": 1, "subscriptionObserverMap": [Map]}}, "event": "ConnectionStateChange", "message": "Connection state is Connected"}
 LOG  [DEBUG] 19:08.989 Hub - Dispatching to api with  {"data": {"connectionState": "Connected", "provider": {"_config": [Object], "awsRealTimeSocket": [WebSocket], "connectionState": "Connecting", "connectionStateMonitor": [ConnectionStateMonitor], "connectionStateMonitorSubscription": [Subscription], "keepAliveAlertTimeoutId": 738, "keepAliveTimeout": 300000, "keepAliveTimeoutId": 737, "promiseArray": [Array], "reconnectionMonitor": [ReconnectionMonitor], "socketStatus": 1, "subscriptionObserverMap": [Map]}}, "event": "ConnectionStateChange", "message": "Connection state is Connected"}
 LOG  [DEBUG] 19:08.996 Util - error on  [TypeError: null is not an object (evaluating '_c[opName].items')]
 LOG  [DEBUG] 19:08.997 Util -  retrying in 294.3436297171284 ms
 LOG  [DEBUG] 19:09.29 AWSAppSyncRealTimeProvider - subscription message from AWS AppSync RealTime: {"id":"208f5d09-5e48-49db-ab35-984043e18eb7","type":"start_ack"}
 LOG  [DEBUG] 19:09.30 AWSAppSyncRealTimeProvider {"id": "208f5d09-5e48-49db-ab35-984043e18eb7", "observer": {"_subscription": {"_cleanup": [Function anonymous], "_observer": [Object], "_queue": undefined, "_state": "ready"}}, "query": "subscription operation {
  onUpdateTagRecord {
    id
    tagId
    dateTime
    forkLength
    species
    latitude
    longitude
    photoBase64
    createdAt
    updatedAt
    _version
    _lastChangedAt
    _deleted
  }
}
", "variables": {}}
 LOG  [DEBUG] 19:09.30 AWSAppSyncRealTimeProvider - subscription ready for {"query":"subscription operation {\n  onUpdateTagRecord {\n    id\n    tagId\n    dateTime\n    forkLength\n    species\n    latitude\n    longitude\n    photoBase64\n    createdAt\n    updatedAt\n    _version\n    _lastChangedAt\n    _deleted\n  }\n}\n","variables":{}}
 LOG  [DEBUG] 19:09.158 AWSAppSyncRealTimeProvider - subscription message from AWS AppSync RealTime: {"id":"eff1c641-a352-4cea-9796-9fa246473c29","type":"start_ack"}
 LOG  [DEBUG] 19:09.159 AWSAppSyncRealTimeProvider {"id": "eff1c641-a352-4cea-9796-9fa246473c29", "observer": {"_subscription": {"_cleanup": [Function anonymous], "_observer": [Object], "_queue": undefined, "_state": "ready"}}, "query": "subscription operation {
  onDeleteTagRecord {
    id
    tagId
    dateTime
    forkLength
    species
    latitude
    longitude
    photoBase64
    createdAt
    updatedAt
    _version
    _lastChangedAt
    _deleted
  }
}
", "variables": {}}
 LOG  [DEBUG] 19:09.162 AWSAppSyncRealTimeProvider - subscription ready for {"query":"subscription operation {\n  onDeleteTagRecord {\n    id\n    tagId\n    dateTime\n    forkLength\n    species\n    latitude\n    longitude\n    photoBase64\n    createdAt\n    updatedAt\n    _version\n    _lastChangedAt\n    _deleted\n  }\n}\n","variables":{}}
 LOG  [DEBUG] 19:09.305 Util -  attempt #2 with this vars: ["query operation($limit: Int, $nextToken: String, $lastSync: AWSTimestamp, $filter: ModelTagRecordFilterInput){\n\t\tsyncTagRecords(limit: $limit, nextToken: $nextToken, lastSync: $lastSync, filter: $filter){\n\t\t\titems {\n\t\t\t\t\t\t\tid\ntagId\ndateTime\nforkLength\nspecies\nlatitude\nlongitude\nphotoBase64\ncreatedAt\nupdatedAt\n_version\n_lastChangedAt\n_deleted\n\t\t\t\t\t\t}\n\t\t\t\t\t\tnextToken\n\t\t\t\t\t\tstartedAt\n\t\t}\n\t}",{"limit":1000,"nextToken":null,"lastSync":0,"filter":null}]
 LOG  [DEBUG] 19:09.311 RestClient - POST https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql
 LOG  [DEBUG] 19:09.566 Util - error on  [TypeError: null is not an object (evaluating '_c[opName].items')]
 LOG  [DEBUG] 19:09.568 Util -  retrying in 491.270243559188 ms
 LOG  [DEBUG] 19:10.73 Util -  attempt #3 with this vars: ["query operation($limit: Int, $nextToken: String, $lastSync: AWSTimestamp, $filter: ModelTagRecordFilterInput){\n\t\tsyncTagRecords(limit: $limit, nextToken: $nextToken, lastSync: $lastSync, filter: $filter){\n\t\t\titems {\n\t\t\t\t\t\t\tid\ntagId\ndateTime\nforkLength\nspecies\nlatitude\nlongitude\nphotoBase64\ncreatedAt\nupdatedAt\n_version\n_lastChangedAt\n_deleted\n\t\t\t\t\t\t}\n\t\t\t\t\t\tnextToken\n\t\t\t\t\t\tstartedAt\n\t\t}\n\t}",{"limit":1000,"nextToken":null,"lastSync":0,"filter":null}]
 LOG  [DEBUG] 19:10.81 RestClient - POST https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql
 LOG  [DEBUG] 19:10.414 Util - error on  [TypeError: null is not an object (evaluating '_c[opName].items')]
 LOG  [DEBUG] 19:10.415 Util -  retrying in 877.3496723508604 ms
 LOG  [DEBUG] 19:11.309 Util -  attempt #4 with this vars: ["query operation($limit: Int, $nextToken: String, $lastSync: AWSTimestamp, $filter: ModelTagRecordFilterInput){\n\t\tsyncTagRecords(limit: $limit, nextToken: $nextToken, lastSync: $lastSync, filter: $filter){\n\t\t\titems {\n\t\t\t\t\t\t\tid\ntagId\ndateTime\nforkLength\nspecies\nlatitude\nlongitude\nphotoBase64\ncreatedAt\nupdatedAt\n_version\n_lastChangedAt\n_deleted\n\t\t\t\t\t\t}\n\t\t\t\t\t\tnextToken\n\t\t\t\t\t\tstartedAt\n\t\t}\n\t}",{"limit":1000,"nextToken":null,"lastSync":0,"filter":null}]
 LOG  [DEBUG] 19:11.316 RestClient - POST https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql
 LOG  [DEBUG] 19:11.601 Util - error on  [TypeError: null is not an object (evaluating '_c[opName].items')]
 LOG  [DEBUG] 19:11.603 Util -  retrying in 1688.0260349438265 ms
 LOG  [DEBUG] 19:13.293 Util -  attempt #5 with this vars: ["query operation($limit: Int, $nextToken: String, $lastSync: AWSTimestamp, $filter: ModelTagRecordFilterInput){\n\t\tsyncTagRecords(limit: $limit, nextToken: $nextToken, lastSync: $lastSync, filter: $filter){\n\t\t\titems {\n\t\t\t\t\t\t\tid\ntagId\ndateTime\nforkLength\nspecies\nlatitude\nlongitude\nphotoBase64\ncreatedAt\nupdatedAt\n_version\n_lastChangedAt\n_deleted\n\t\t\t\t\t\t}\n\t\t\t\t\t\tnextToken\n\t\t\t\t\t\tstartedAt\n\t\t}\n\t}",{"limit":1000,"nextToken":null,"lastSync":0,"filter":null}]
 LOG  [DEBUG] 19:13.300 RestClient - POST https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql
 LOG  [DEBUG] 19:13.564 Util - error on  [TypeError: null is not an object (evaluating '_c[opName].items')]
 LOG  [DEBUG] 19:13.565 Util -  retrying in 3205.637959793806 ms
 LOG  [DEBUG] 19:16.773 Util -  attempt #6 with this vars: ["query operation($limit: Int, $nextToken: String, $lastSync: AWSTimestamp, $filter: ModelTagRecordFilterInput){\n\t\tsyncTagRecords(limit: $limit, nextToken: $nextToken, lastSync: $lastSync, filter: $filter){\n\t\t\titems {\n\t\t\t\t\t\t\tid\ntagId\ndateTime\nforkLength\nspecies\nlatitude\nlongitude\nphotoBase64\ncreatedAt\nupdatedAt\n_version\n_lastChangedAt\n_deleted\n\t\t\t\t\t\t}\n\t\t\t\t\t\tnextToken\n\t\t\t\t\t\tstartedAt\n\t\t}\n\t}",{"limit":1000,"nextToken":null,"lastSync":0,"filter":null}]
 LOG  [DEBUG] 19:16.781 RestClient - POST https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql
 LOG  [DEBUG] 19:17.6 Util - error on  [TypeError: null is not an object (evaluating '_c[opName].items')]
 LOG  [DEBUG] 19:17.8 Util -  retrying in 6417.742432202537 ms
 LOG  [DEBUG] 19:23.429 Util -  attempt #7 with this vars: ["query operation($limit: Int, $nextToken: String, $lastSync: AWSTimestamp, $filter: ModelTagRecordFilterInput){\n\t\tsyncTagRecords(limit: $limit, nextToken: $nextToken, lastSync: $lastSync, filter: $filter){\n\t\t\titems {\n\t\t\t\t\t\t\tid\ntagId\ndateTime\nforkLength\nspecies\nlatitude\nlongitude\nphotoBase64\ncreatedAt\nupdatedAt\n_version\n_lastChangedAt\n_deleted\n\t\t\t\t\t\t}\n\t\t\t\t\t\tnextToken\n\t\t\t\t\t\tstartedAt\n\t\t}\n\t}",{"limit":1000,"nextToken":null,"lastSync":0,"filter":null}]
 LOG  [DEBUG] 19:23.438 RestClient - POST https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql
 LOG  [DEBUG] 19:23.725 Util - error on  [TypeError: null is not an object (evaluating '_c[opName].items')]
 LOG  [DEBUG] 19:23.726 Util -  retrying in 12839.747941535315 ms
 LOG  [DEBUG] 19:36.568 Util -  attempt #8 with this vars: ["query operation($limit: Int, $nextToken: String, $lastSync: AWSTimestamp, $filter: ModelTagRecordFilterInput){\n\t\tsyncTagRecords(limit: $limit, nextToken: $nextToken, lastSync: $lastSync, filter: $filter){\n\t\t\titems {\n\t\t\t\t\t\t\tid\ntagId\ndateTime\nforkLength\nspecies\nlatitude\nlongitude\nphotoBase64\ncreatedAt\nupdatedAt\n_version\n_lastChangedAt\n_deleted\n\t\t\t\t\t\t}\n\t\t\t\t\t\tnextToken\n\t\t\t\t\t\tstartedAt\n\t\t}\n\t}",{"limit":1000,"nextToken":null,"lastSync":0,"filter":null}]
 LOG  [DEBUG] 19:36.576 RestClient - POST https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql
 LOG  [DEBUG] 19:37.194 Util - error on  [TypeError: null is not an object (evaluating '_c[opName].items')]
 LOG  [DEBUG] 19:37.195 Util -  retrying in 25630.7610649155 ms
 LOG  [DEBUG] 20:02.829 Util -  attempt #9 with this vars: ["query operation($limit: Int, $nextToken: String, $lastSync: AWSTimestamp, $filter: ModelTagRecordFilterInput){\n\t\tsyncTagRecords(limit: $limit, nextToken: $nextToken, lastSync: $lastSync, filter: $filter){\n\t\t\titems {\n\t\t\t\t\t\t\tid\ntagId\ndateTime\nforkLength\nspecies\nlatitude\nlongitude\nphotoBase64\ncreatedAt\nupdatedAt\n_version\n_lastChangedAt\n_deleted\n\t\t\t\t\t\t}\n\t\t\t\t\t\tnextToken\n\t\t\t\t\t\tstartedAt\n\t\t}\n\t}",{"limit":1000,"nextToken":null,"lastSync":0,"filter":null}]
 LOG  [DEBUG] 20:02.838 RestClient - POST https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql
 LOG  [DEBUG] 20:03.106 Util - error on  [TypeError: null is not an object (evaluating '_c[opName].items')]
 LOG  [DEBUG] 20:03.108 Util -  retrying in 51289.37299061342 ms
 LOG  [DEBUG] 20:08.798 AWSAppSyncRealTimeProvider - subscription message from AWS AppSync RealTime: {"type":"ka"}
 LOG  [DEBUG] 20:08.799 AWSAppSyncRealTimeProvider {"id": "", "observer": null, "query": "", "variables": {}}
 LOG  [DEBUG] 20:54.400 Util -  attempt #10 with this vars: ["query operation($limit: Int, $nextToken: String, $lastSync: AWSTimestamp, $filter: ModelTagRecordFilterInput){\n\t\tsyncTagRecords(limit: $limit, nextToken: $nextToken, lastSync: $lastSync, filter: $filter){\n\t\t\titems {\n\t\t\t\t\t\t\tid\ntagId\ndateTime\nforkLength\nspecies\nlatitude\nlongitude\nphotoBase64\ncreatedAt\nupdatedAt\n_version\n_lastChangedAt\n_deleted\n\t\t\t\t\t\t}\n\t\t\t\t\t\tnextToken\n\t\t\t\t\t\tstartedAt\n\t\t}\n\t}",{"limit":1000,"nextToken":null,"lastSync":0,"filter":null}]
 LOG  [DEBUG] 20:54.409 RestClient - POST https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql
 LOG  [DEBUG] 20:54.683 Util - error on  [TypeError: null is not an object (evaluating '_c[opName].items')]
 LOG  [DEBUG] 20:54.685 Util -  retrying in 102485.6777054331 ms
 LOG  [DEBUG] 21:08.800 AWSAppSyncRealTimeProvider - subscription message from AWS AppSync RealTime: {"type":"ka"}
 LOG  [DEBUG] 21:08.802 AWSAppSyncRealTimeProvider {"id": "", "observer": null, "query": "", "variables": {}}
 LOG  [DEBUG] 22:08.800 AWSAppSyncRealTimeProvider - subscription message from AWS AppSync RealTime: {"type":"ka"}
 LOG  [DEBUG] 22:08.806 AWSAppSyncRealTimeProvider {"id": "", "observer": null, "query": "", "variables": {}}
 LOG  [DEBUG] 22:37.173 Util -  attempt #11 with this vars: ["query operation($limit: Int, $nextToken: String, $lastSync: AWSTimestamp, $filter: ModelTagRecordFilterInput){\n\t\tsyncTagRecords(limit: $limit, nextToken: $nextToken, lastSync: $lastSync, filter: $filter){\n\t\t\titems {\n\t\t\t\t\t\t\tid\ntagId\ndateTime\nforkLength\nspecies\nlatitude\nlongitude\nphotoBase64\ncreatedAt\nupdatedAt\n_version\n_lastChangedAt\n_deleted\n\t\t\t\t\t\t}\n\t\t\t\t\t\tnextToken\n\t\t\t\t\t\tstartedAt\n\t\t}\n\t}",{"limit":1000,"nextToken":null,"lastSync":0,"filter":null}]
 LOG  [DEBUG] 22:37.178 RestClient - POST https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql
 LOG  [DEBUG] 22:37.881 Util - error on  [TypeError: null is not an object (evaluating '_c[opName].items')]
 LOG  [DEBUG] 22:37.884 Util -  retrying in 204887.13716594744 ms
 LOG  [DEBUG] 23:08.803 AWSAppSyncRealTimeProvider - subscription message from AWS AppSync RealTime: {"type":"ka"}
 LOG  [DEBUG] 23:08.805 AWSAppSyncRealTimeProvider {"id": "", "observer": null, "query": "", "variables": {}}
 LOG  [DEBUG] 24:08.802 AWSAppSyncRealTimeProvider - subscription message from AWS AppSync RealTime: {"type":"ka"}
 LOG  [DEBUG] 24:08.804 AWSAppSyncRealTimeProvider {"id": "", "observer": null, "query": "", "variables": {}}
 LOG  [DEBUG] 25:08.799 AWSAppSyncRealTimeProvider - subscription message from AWS AppSync RealTime: {"type":"ka"}
 LOG  [DEBUG] 25:08.801 AWSAppSyncRealTimeProvider {"id": "", "observer": null, "query": "", "variables": {}}

```

</details>

### aws-exports.js

```javascript
/* eslint-disable */
// WARNING: DO NOT EDIT. This file is automatically generated by AWS Amplify. It will be overwritten.

const awsmobile = {
    "aws_project_region": "us-west-2",
    "aws_appsync_graphqlEndpoint": "https://XXXXX.appsync-api.us-west-2.amazonaws.com/graphql",
    "aws_appsync_region": "us-west-2",
    "aws_appsync_authenticationType": "API_KEY",
    "aws_appsync_apiKey": "XXXXX"
};


export default awsmobile;

```

### Mobile Device

iPhone 12 mini

### Mobile Operating System

iOS 16.3

**Comment by jaimemarijke** [comment]
Update: I just tried leaving my app running with the debugger on in the background and I finally saw it sync and got these logs:

```
 LOG  [DEBUG] 49:57.468 RestClient - POST https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql
 LOG  [DEBUG] 49:57.826 Util - error on  [TypeError: null is not an object (evaluating '_c[opName].items')]
 LOG  [DEBUG] 49:57.829 Util -  retrying in 204829.51192430555 ms
 LOG  [DEBUG] 50:28.902 AWSAppSyncRealTimeProvider - subscription message from AWS AppSync RealTime: {"type":"ka"}
 LOG  [DEBUG] 50:28.902 AWSAppSyncRealTimeProvider {"id": "", "observer": null, "query": "", "variables": {}}
 LOG  [DEBUG] 51:28.872 AWSAppSyncRealTimeProvider - subscription message from AWS AppSync RealTime: {"type":"ka"}
 LOG  [DEBUG] 51:28.874 AWSAppSyncRealTimeProvider {"id": "", "observer": null, "query": "", "variables": {}}
 LOG  [DEBUG] 52:28.873 AWSAppSyncRealTimeProvider - subscription message from AWS AppSync RealTime: {"type":"ka"}
 LOG  [DEBUG] 52:28.876 AWSAppSyncRealTimeProvider {"id": "", "observer": null, "query": "", "variables": {}}
 LOG  [DEBUG] 53:22.663 Util -  attempt #12 with this vars: ["query operation($limit: Int, $nextToken: String, $lastSync: AWSTimestamp, $filter: ModelTagRecordFilterInput){\n\t\tsyncTagRecords(limit: $limit, nextToken: $nextToken, lastSync: $lastSync, filter: $filter){\n\t\t\titems {\n\t\t\t\t\t\t\tid\ntagId\ndateTime\nforkLength\nspecies\nlatitude\nlongitude\nphotoBase64\ncreatedAt\nupdatedAt\n_version\n_lastChangedAt\n_deleted\n\t\t\t\t\t\t}\n\t\t\t\t\t\tnextToken\n\t\t\t\t\t\tstartedAt\n\t\t}\n\t}",{"limit":1000,"nextToken":null,"lastSync":0,"filter":null}]
 LOG  [DEBUG] 53:22.675 RestClient - POST https://vnyaearw6zcbllzmjv2ti2w75u.appsync-api.us-west-2.amazonaws.com/graphql
 LOG  [DEBUG] 53:23.137 Util - error on  [TypeError: null is not an object (evaluating '_c[opName].items')]
 LOG  [DEBUG] 53:23.138 Util -  retrying in false ms
 LOG  [DEBUG] 53:23.142 DataStore - Sync failed with authMode: API_KEY [TypeError: null is not an object (evaluating '_c[opName].items')]
 WARN  [WARN] 53:23.143 DataStore - User is unauthorized to query syncTagRecords with auth mode API_KEY. No data could be returned.
 LOG  [DEBUG] 53:23.205 DataStore - Next fullSync in 86399.966 seconds. (Sat Apr 01 2023 16:53:23 GMT-0700 (Pacific Daylight Time))
 LOG  [DEBUG] 53:23.210 DataStore - Attempting mutation with authMode: API_KEY
 LOG  [DEBUG] 53:23.211 Util -  attempt #1 with this vars: ["TagRecord","Create","{\"id\":\"0ca08f49-2d03-4ac7-bb64-4b7cafd10327\",\"tagId\":\"HT-12345-Stub\",\"dateTime\":\"2023-03-31T23:46:17.151Z\",\"forkLength\":4.46,\"species\":\"Bluefin Tuna\",\"latitude\":48.74,\"longitude\":-122.447,\"photoBase64\":\"<REDACTED>",\"createdAt\":null,\"updatedAt\":null}","{}",null,null,{"data":"{\"id\":\"0ca08f49-2d03-4ac7-bb64-4b7cafd10327\",\"tagId\":\"HT-12345-Stub\",\"dateTime\":\"2023-03-31T23:46:17.151Z\",\"forkLength\":4.46,\"species\":\"Bluefin Tuna\",\"latitude\":48.74,\"longitude\":-122.447,\"photoBase64\":\"<REDACTED>","createdAt":"2023-03-31T23:53:23.437Z","updatedAt":"2023-03-31T23:53:23.437Z","_version":1,"_lastChangedAt":1680306803440,"_deleted":null}}}}
 LOG  [DEBUG] 53:23.882 AWSAppSyncRealTimeProvider {"id": "5159c8dd-8abd-46e0-bf98-921910a046f4", "observer": {"_subscription": {"_cleanup": [Function anonymous], "_observer": [Object], "_queue": undefined, "_state": "ready"}}, "query": "subscription operation {
  onCreateTagRecord {
    id
    tagId
    dateTime
    forkLength
    species
    latitude
    longitude
    photoBase64
    createdAt
    updatedAt
    _version
    _lastChangedAt
    _deleted
  }
}
", "variables": {}}
 LOG  [DEBUG] 53:28.872 AWSAppSyncRealTimeProvider - subscription message from AWS AppSync RealTime: {"type":"ka"}
 LOG  [DEBUG] 53:28.873 AWSAppSyncRealTimeProvider {"id": "", "observer": null, "query": "", "variables": {}}
 LOG  [DEBUG] 54:28.873 AWSAppSyncRealTimeProvider - subscription message from AWS AppSync RealTime: {"type":"ka"}
 LOG  [DEBUG] 54:28.875 AWSAppSyncRealTimeProvider {"id": "", "observer": null, "query": "", "variables": {}}
 LOG  [DEBUG] 55:28.872 AWSAppSyncRealTimeProvider - subscription message from AWS AppSync RealTime: {"type":"ka"}
 LOG  [DEBUG] 55:28.873 AWSAppSyncRealTimeProvider {"id": "", "observer": null, "query": "", "variables": {}}
```

**Comment by jaimemarijke** [comment]
Well, this is now fixed for me, and it seems to be directly related to some changes I made, but I'm a little mystified at the connection.

I think the DataStore issue might have ended up being something to do with a schema change causing rows of data in the DynamoDB being unreadable by Amplify. A while ago, I had made a schema change to my data model (I removed a field, and I changed another field from an Enum to String type). I had saved & deployed the change, and pulled locally, but I think I must have forgotten to commit the schema changes to my repo. 

What I did that fixed it:
- ran `amplify pull` to get my schema changes back in the repo
- deleted all the rows in my DynamoDB (thankfully this is still in development, so I'm free to just do that).

After that new records started showing up in Amplify Studio again.

I'm pretty confused why a mismatch schema would cause Amplify to throw authorization errors. My experience trying to debug this issue has made me inclined not to use AWS Amplify in future projects, and I'll probably port off of it before releasing my current project to production.

**Comment by chrisbonifacio** [comment]
Hi @jaimemarijke 👋 apologies for the delay.

Glad you were able to unblock yourself! 🎉 

When there is a schema mismatch or if there are records that do not pass schema validation during sync they will cause errors in the sync process.

For example, say you have a `Post` model, with an optional `tags: [String]` field. On creation it can be `null`.
DataStore will sync those records just fine, as long as any other required fields have values.

Where things might break is if `tags` is made required, or "non-nullable".
Now, DataStore will fail to sync the Post model records with an error similar to: `Cannot return null for non-nullable type: 'String' within parent 'Post` because some records will return `null` for a "non-nullable" field.

The solution to this is to either backfill existing data with a default value for the field in production (leveraging the [@default directive](https://docs.amplify.aws/cli/graphql/data-modeling/#assign-default-values-for-fields) can help to mitigate this), or in a development environment it would be safe to simply clear the database and start fresh with no invalid data.

This might explain why clearing the database seems to have fixed a sync issue.

I would also recommend enabling [multi-auth](https://docs.amplify.aws/lib/datastore/setup-auth-rules/q/platform/js/#configure-multiple-authorization-types) for DataStore if you haven't already done so, without it enabled DataStore will only use one auth mode which can result in red herrings and make it seem like sync's not working when it's just warning that some records can't be synced with the current auth mode.

This should help mitigate most sync problems.

In any case, should you decide to keep using Amplify, and I hope you do because it can be a great tool for iterating quickly once you get the ball rolling.

If the learning curve for DataStore is hindering progress and/or you don't have a hard requirement on offline capabilities, I would give Amplify's `API.graphql` a try as it will give you more control over the shape of the graphql queries, auth modes, any have access to some features DataStore currently does not such as dynamic owner and group auth.
